install_config/topics/configuring_for_aws_registry-proc.adoc

```diff
@@ -0,0 +1,293 @@
+openshift_hosted_registry_storage_s3_accesskey=AKIAJ6VLREDHATSPBUA <1>
+openshift_hosted_registry_storage_s3_secretkey=g/8PmTYDQVGssFWWFvfawHpDbZyGkjGNZhbWQpjH <2>
+openshift_hosted_registry_storage_s3_bucket=openshift-registry-storage <3>
+openshift_hosted_registry_storage_s3_region=us-east-1 <4>
+openshift_hosted_registry_storage_s3_chunksize=26214400
+openshift_hosted_registry_storage_s3_rootdirectory=/registry
+openshift_hosted_registry_pullthrough=true
+openshift_hosted_registry_acceptschema2=true
+openshift_hosted_registry_enforcequota=true
+openshift_hosted_registry_replicas=3
+----
+<1> The access key for the IAM user.
+<2> The secret key for the IAM user.
+<3> The S3 storage bucket name.
+<4> The region in which the bucket exists.
+
+
+=== Manually configuring {product-title} registry to use S3
+
+To use Amazon Web Services (AWS) S3 object storage, edit the registry’s configuration file and mount to the registry pod.
+
+.Procedure
+
+. Export the current *_config.yml_*:
++
+[source,bash]
+----
+$ oc get secret registry-config \
+    -o jsonpath='{.data.config\.yml}' -n default | base64 -d \
+  >> config.yml.old
+----
+
+. Create a new configuration file from the old *_config.yml_*:
++
+[source,bash]
+----
+$ cp config.yml.old config.yml
+----
+
+. Edit the file to include the S3 parameters. Specify the accountname, accountkey,
+container, and realm in the `storage` section of a registry’s configuration file:
++
+[source,yaml]
+----
+storage:
+  delete:
+    enabled: true
+  cache:
+    blobdescriptor: inmemory
+  s3:
+    accesskey: AKIAJ6VLREDHATSPBUA
+    secretkey: g/8PmTYDQVGssFWWFvfawHpDbZyGkjGNZhbWQpjH
+    region: us-east-1
+    bucket: openshift-registry-storage
+    encrypt: False
+    secure: true
+    v4auth: true
+    rootdirectory: /registry
+    chunksize: "26214400"
+----
+<1>	Replace with the storage account name.
+<2> The account key associated to the <account-name>.
+<3> Directory used to store the data. `registry` by default
+<4> Storage realm `core.windows.net` by default
+
+. Delete the `registry-config` secret:
++
+[source,bash]
+----
+$ oc delete secret registry-config -n default
+----
+
+. Recreate the configmap to reference the updated configuration file:
++
+[source,bash]
+----
+$ oc create secret generic registry-config \
+    --from-file=config.yml -n default
+----
+
+. Redeploy the registry to read the updated configuration:
++
+[source,bash]
+----
+$ oc rollout latest docker-registry -n default
+----
+
+=== Verify the registry is using S3 storage
+
+To verify if the registry is using Amazon S3 storage:
+
+.Procedure
+
+. After a successful registry deployment, the registry `deploymentconfig` does
+not show any information if the registry is using an `emptydir` instead of AWS S3
+storage:
++
+[source,bash]
+----
+$ oc describe dc docker-registry -n default
+...
+Mounts:
+  ...
+  /registry from registry-storage (rw)
+Volumes:
+registry-storage:
+Type:       EmptyDir <1>
+...
```

Comment:
If s3 storage is mounted as secret with steps above, type should not be EmptyDir here, it should be like below:
   volume-pwc2g:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	registry-config
    Optional:	false

Reply:
@wzheng1 here is the output using 3.11 with s3. Do you think it would be helpful if i just included more information here. It still states Type: EmptyDir but the secret defines the S3 values.

```
[root@ip-10-10-1-60 ~]# oc describe dc docker-registry -n default
Name:		docker-registry
Namespace:	default
Created:	36 minutes ago
Labels:		docker-registry=default
Annotations:	<none>
Latest Version:	1
Selector:	docker-registry=default
Replicas:	1
Triggers:	Config
Strategy:	Rolling
Template:
Pod Template:
  Labels:		docker-registry=default
  Service Account:	registry
  Containers:
   registry:
    Image:	registry.redhat.io/openshift3/ose-docker-registry:v3.11.16
    Port:	5000/TCP
    Host Port:	0/TCP
    Requests:
      cpu:	100m
      memory:	256Mi
    Liveness:	http-get https://:5000/healthz delay=10s timeout=5s period=10s #success=1 #failure=3
    Readiness:	http-get https://:5000/healthz delay=0s timeout=5s period=10s #success=1 #failure=3
    Environment:
      REGISTRY_HTTP_ADDR:					:5000
      REGISTRY_HTTP_NET:					tcp
      REGISTRY_HTTP_SECRET:					SPLR83SDsPaGbGuwSMDfnDwrDRvGf6YXl4h9JQrToQU=
      REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_ENFORCEQUOTA:	false
      REGISTRY_HTTP_TLS_KEY:					/etc/secrets/registry.key
      OPENSHIFT_DEFAULT_REGISTRY:				docker-registry.default.svc:5000
      REGISTRY_CONFIGURATION_PATH:				/etc/registry/config.yml
      REGISTRY_OPENSHIFT_SERVER_ADDR:				docker-registry.default.svc:5000
      REGISTRY_HTTP_TLS_CERTIFICATE:				/etc/secrets/registry.crt
    Mounts:
      /etc/registry from docker-config (rw)
      /etc/secrets from registry-certificates (rw)
      /registry from registry-storage (rw)
  Volumes:
   registry-storage:
    Type:	EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:	
   registry-certificates:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	registry-certificates
    Optional:	false
   docker-config:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	registry-config
    Optional:	false
```

Reply:
Yes, it still can display EmptyDir, but as you can see it defines REGISTRY_CONFIGURATION_PATH:				/etc/registry/config.yml, so I suggest we can still list EmptyDir, but add this part to it and clarify the docker registry actually read from here, WDYT?
docker-config:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	registry-config
    Optional:	false

Reply:
@wzheng1 I believe I covered this correctly in the latest patch. Let me know if it works.

Reply:
@cooktheryan Thanks! LGTM now : )